localstack/utils/no_exit_argument_parser.py

```diff
@@ -0,0 +1,17 @@
+import argparse
+import logging
+from typing import NoReturn
+
+LOG = logging.getLogger(__name__)
+
+
+# Implements the `exit_on_error=False` behavior introduced in Python 3.9 to support older Python versions
+# and prevents further SystemExit for other error categories.
+# Limitations of error cases: https://stackoverflow.com/a/67891066/6875981
+# Subclassing workaround example: https://stackoverflow.com/a/59072378/6875981
+class NoExitArgumentParser(argparse.ArgumentParser):
+    def exit(self, status: int = ..., message: str | None = ...) -> NoReturn:
+        LOG.warning(f"Error in argument parser but preventing exit: {message}")
```

Comment:
I guess that's not even needed 🤔 but better be safe than exiting here.

Reply:
What exactly is not needed and why? The subclassing workaround is not needed due to no required arguments?

Reply:
ups sorry, the anchor is not that clear on GH.

I just wanted to refer to the `exit` method overriding. It appears that exit is never invoked so far but I thought it is better to protect us from an accidental `sys. exit()` (default behavior of argparse for CLI).

The subclassing workaround is definitely required because even the [exit_on_error](https://docs.python.org/3/library/argparse.html#exit-on-error) flag added in Python 3.9 only catches a subset of argument errors and just exits in other cases.